src/util.h

```diff
@@ -201,27 +201,30 @@ template <typename Inner, typename Outer>
 constexpr ContainerOfHelper<Inner, Outer> ContainerOf(Inner Outer::*field,
                                                       Inner* pointer);
 
-// If persistent.IsWeak() == false, then do not call persistent.Reset()
-// while the returned Local<T> is still in scope, it will destroy the
-// reference to the object.
-template <class TypeName>
-inline v8::Local<TypeName> PersistentToLocal(
-    v8::Isolate* isolate,
-    const Persistent<TypeName>& persistent);
-
-// Unchecked conversion from a non-weak Persistent<T> to Local<T>,
-// use with care!
-//
-// Do not call persistent.Reset() while the returned Local<T> is still in
-// scope, it will destroy the reference to the object.
-template <class TypeName>
-inline v8::Local<TypeName> StrongPersistentToLocal(
-    const Persistent<TypeName>& persistent);
-
-template <class TypeName>
-inline v8::Local<TypeName> WeakPersistentToLocal(
-    v8::Isolate* isolate,
-    const Persistent<TypeName>& persistent);
+class PersistentToLocal {
```

Comment:
Shouldn't this be put into `node_persistent.h`? Since these are all `node::Persistent`

Reply:
Why a `class` and not a `namespace`? they are all static templates?

Reply:
P.S. why not move the implementations here? They are all single lines (some could even be `constexpr` on C++17 compilers)

Reply:
> Doesn't that bring more complexity than necessary? (the constexpr approach)

A little bit, yes. So up to author. BTW I think there's a CONSTEXPR macro defined by V8 based of this condition, so we could reuse.

> How is it different from doing `process.versions.v8` detection in JS and use harmony features?

It's compile time only, 0 run time cost, only benefits. (only cost is the above mentioned code complexity)

Reply:
> Shouldn't this be put into node_persistent.h? Since these are all node::Persistent

You mean as `node::Persistent` methods? I'd say that's the best/most logical place for it.

Reply:
@bnoordhuis I tried importing it as `node::Persistent` methods, but `node::Persistent` is merely an [alias](https://github.com/nodejs/node/blob/master/src/node_persistent.h#L24) to `v8::Persistent`, so I can't really extend it, AFAIK.